pennylane/qnode.py

```diff
@@ -703,6 +707,8 @@ def check_method(m):
             else:
                 y0 = None
 
+        variances = any(e.return_type == 'variance' for e in self.ev)
```

Comment:
This is because the gradient for loop loops over _parameters_, not measurements. It could be the case that a single parameter affects both expectation values and variances. So if there is at least one variance in the return list, we need to instead call `_pd_analytic_var`, which knows how to deal with a mix of expectations and variances.